docs/pages/application-access/cloud-apis/google-cloud.mdx

```diff
@@ -0,0 +1,615 @@
+
+Verify that your Teleport user can see the `google-cloud-cli` application you
+registered earlier:
+
+```code
+$ tsh apps ls
+Application      Description Type Public Address                        Labels
+---------------- ----------- ---- ------------------------------------- -------------------
+google-cloud-cli             HTTP google-cloud-cli.teleport.example.com teleport.dev/origin
+```
+
+### Log in to use a Google Cloud CLI
+
+Log in to the application, specifying that you would like to assume the
+`teleport-vm-lister` service account:
+
+```code
+$ tsh app login google-cloud-cli --gcp-service-account teleport-vm-lister
+```
+
+This command validates the value of the `--gcp-service-account` flag against the
+ones the user is authorized to assume. The value of the flag can either be the
+full URI of the service account or the name of the identity, e.g.,
+`teleport-vm-lister`. 
+
+A user can omit the `--gcp-service-account` flag if they are only authorized to
+access a single Google Cloud service account, but otherwise an empty
+`--gcp-service-account` flag will result in an error.
+
+If the command succeeds, you will see information about the user's chosen Google
+Cloud service account similar to the following:
+
+```text
+Logged into GCP app "google-cloud-cli".
+Your service account: teleport-google-cloud-cli@my-project.iam.gserviceaccount.com
+Example command: tsh gcloud compute instances list
+```
+
+### Execute Google Cloud CLI commands
+
+At this point, you can run `gcloud` commands using the Teleport Application
+Service by prefixing them with `tsh`. Since your user authenticated to your
+Google Cloud CLI application with a service account that can list VMs, for
+example, run this command to do so: 
+
+```code
+$ tsh gcloud compute instances list
+```
+
+You should see a list of virtual machines in your Google Cloud project.
+
+However, your Teleport user cannot create a VM, since its service account
+does not have this authorization:
+
+```code
+$ tsh gcloud compute instances create another-instance --zone=<Var name="google-cloud-zone" />
+ERROR: (gcloud.compute.instances.create) Could not fetch resource:
+ - Required 'compute.instances.create' permission for 'projects/my-project/zones/my-zone/instances/another-instance'
+
+ERROR: exit status 1
+```
+
+### Use Google Cloud CLI applications without `tsh`
+
+In addition to running `gcloud` commands via `tsh`, you can grant secure access
+to any CLI application that executes commands against Google Cloud's APIs.
+
+To do this, use `tsh` to start a local proxy that forwards traffic from your CLI
+application to the Teleport Application Service. The Application Service uses
+the `teleport-google-cloud-cli` service account we created earlier to fetch an
+authentication token from Google Cloud. Your CLI application uses this token to
+authenticate requests to Google Cloud's APIs.
+
+To start the local proxy, run the following `tsh` command:
+
+```code
+$ tsh proxy gcloud 
```

Comment:
I wonder if we should rename the proxy command to `tsh proxy gcp` instead, as GCP is the cloud and `gcloud` is one of the command line tools. Thoughts?

Reply:
Good question—it looks like Google has been moving away from the "GCP" usage in favor of "Google Cloud" (e.g., on the homepage: https://cloud.google.com/), but it seems like "GCP" is still well known and crops up every now and then in their docs (e.g., the diagram here: https://cloud.google.com/docs/overview#global_regional_and_zonal_resources).

I think `tsh proxy gcp` makes sense.

Reply:
Interesting, I didn't realize that Google seems to be moving away from the "GCP" term. I guess that makes `tsh proxy gcloud` a good name after all. I'll think about adding `tsh proxy gcp` as an alias.

Reply:
Weirdly, I can't find any blog post or similar announcement that there's a formal name change, just pages that used to say "Google Cloud Platform" that now say "Google Cloud"